### DISCUSSION TITLE
how to get a remote crate's SOURCES (adb/ads files)

### DISCUSSION THREAD

**Discussion by nnsite** [alire-project/alire#1738] [Q&A]
I see that alr compiles dependencies immediately, and I find no corresponding source file (say, for gnatcoll-terminal.adb, after with-ing and building GNATCOLL.Coders.Streams) anywhere on my system. But I want to study those sources... How do I tell alr to keep the sources (while using the object files for compilations) of whatever crate it gets (or one in particular) ?
There must be a way, but while reading the manual I can't find it.

**Comment by mosteo** [comment]
Sources are not deleted, you can see all paths to sources with `alr printenv` run from inside the root project you're building.

Alternatively, you can `alr get` the crate whose sources you want to study, and it will be downloaded under the current directory.
